tests/filesystem.go

```diff
@@ -319,17 +325,20 @@ func formatPartition(ctx context.Context, partition *types.Partition) error {
 	return nil
 }
 
-func writePartitionData(device string, contents string) error {
+func writePartitionData(device string, contents string) (err error) {
 	bzipped, err := base64.StdEncoding.DecodeString(contents)
 	if err != nil {
 		return err
 	}
 	reader := bzip2.NewReader(bytes.NewBuffer(bzipped))
-	f, err := os.OpenFile(device, os.O_WRONLY, 0644)
+	var f *os.File
+	f, err = os.OpenFile(device, os.O_WRONLY, 0644)
 	if err != nil {
 		return err
 	}
-	defer f.Close()
+	defer func() {
+		err = errors.Join(err, f.Close())
```

Comment:
hmm, this is more then just a join cleanup error, it looks like we are expanding the defer func.  Might should be a part of ignore file close errors commit? a27cf54 and then have the join part be a part of this commit?